virtual_rainforest/core/grid.py

```diff
@@ -1,203 +1,370 @@
-                cell_area=config.cell_area,
-                cell_nx=config.cell_nx,
-                cell_ny=config.cell_ny,
+
+    return cell_dict
+
+
+@register_grid(grid_type="hexagon")
+def make_hex_grid(
+    cell_area: float,
+    cell_nx: int,
+    cell_ny: int,
+    xoff: float = 0,
+    yoff: float = 0,
+) -> dict:
+    """Create a hexagonal grid.
+
+    Args:
+        cell_area: The area of each hexagon cell
+        cell_nx: The number of grid cells in the X direction.
+        cell_ny: The number of grid cells in the Y direction.
+        xoff: An offset to use for the grid origin in the X direction.
+        yoff: An offset to use for the grid origin in the Y direction.
+    """
+
+    # Create the polygon prototype object, with origin at 0,0 and area 1
+    # Note coordinate order is anti-clockwise - right hand rule
+    side_length_a1 = 3 ** (1 / 4) * np.sqrt(2 / 9)
+    apothem_a1 = np.sqrt(3) * side_length_a1 / 2
+    prototype = Polygon(
+        [
+            [apothem_a1, 0],
+            [2 * apothem_a1, side_length_a1 * 0.5],
+            [2 * apothem_a1, side_length_a1 * 1.5],
+            [apothem_a1, side_length_a1 * 2],
+            [0, side_length_a1 * 1.5],
+            [0, side_length_a1 * 0.5],
+            [apothem_a1, 0],
+        ]
+    )
+
+    # Scale to requested size and origin and get side length and apothem for translation
+    # when tiling the grid
+    scale_factor = np.sqrt(cell_area)
+    side_length = 3 ** (1 / 4) * np.sqrt(2 * (cell_area / 9))
+    apothem = np.sqrt(3) * side_length / 2
+    prototype = scale(prototype, xfact=scale_factor, yfact=scale_factor, origin=(0, 0))
+    prototype = translate(prototype, xoff=xoff, yoff=yoff)
+
+    # Tile the prototypes to create the grid
+    cell_dict = {}
+
+    # Loop over columns and rows of cells, incrementing the base coordinates
+    for y_idx in range(cell_ny):
+        for x_idx in range(cell_nx):
+            # Define the cell id as "row_col"
+            cell_id = f"{x_idx}-{y_idx}"
+            # Store shifted geometry under cell id
+            cell_dict[cell_id] = translate(
+                prototype,
+                xoff=2 * apothem * x_idx + apothem * (y_idx % 2),
+                yoff=1.5 * side_length * y_idx,
             )
 
+    return cell_dict
+
+
+@register_grid(grid_type="triangle")
+def make_triangular_grid(
+    cell_area: float,
+    cell_nx: int,
+    cell_ny: int,
+    xoff: float = 0,
+    yoff: float = 0,
+) -> dict:
+    """Create a equilateral triangular grid.
+
+    Args:
+        cell_area: The area of each triangular cell
+        cell_nx: The number of grid cells in the X direction.
+        cell_ny: The number of grid cells in the Y direction.
+        xoff: An offset to use for the grid origin in the X direction.
+        yoff: An offset to use for the grid origin in the Y direction.
+    """
+
+    # Note shapely.affinity.rotate for inversion
+
+    raise NotImplementedError()
+
+
+# @dataclass
```

Comment:
Remove wat is not used - in a separate commit, so it can be recovered easily if needed.